server/src/graphql/resolvers/mutations.js

```diff
@@ -134,8 +139,27 @@ module.exports = {
       model.id = uuidv4();
     }
     try {
-      await db.Well.upsert({ ...model });
-      return db.Well.findByPk(model.id);
+      if (
+        model.status === 'active' ||
+        model.status === 'drilling' ||
+        model.status === 'inactive'
+      ) {
+        if (model.status === 'drilling') {
+          const siblingWells = await db.Well.findAll({
```

Comment:
Would it not be faster to find a single row that has the rigId and the status of drilling. Then based on it that returns anything, conditionally return your error. In general, databases are much more efficient at filtering information than our native algorithms like the for loop